tavern/util/dict_util.py

```diff
@@ -56,26 +52,89 @@ def format_keys(val, variables):
     return formatted
 
 
-def recurse_access_key(current_val, keys):
+def recurse_access_key(data, query):
+    """
+    Search for something in the given data using the given query.
+
+    Note:
+        Falls back to old _recurse_access_key if not found - will be removed in 1.0
+
+    Args:
+        data (dict, list): Data to search in
+        query (str): Query to run
+
+    Returns:
+        object: Whatever was found by the search
+    """
+
+    msg = "In a future version of Tavern, selecting for values to save in nested objects will have to be done as a JMES path query - see http://jmespath.org/ for more information"
+
+    try:
+        from_jmespath = jmespath.search(query, data)
+    except jmespath.exceptions.ParseError:
+        logger.warning("Error parsing JMES query - %s", msg, exc_info=True)
+        from_jmespath = None
+
+    # The value might actually be None, in which case we will search twice for no reason,
+    # but this shouldn't cause any issues
+    if from_jmespath is None:
+        try:
+            from_recurse = _recurse_access_key(data, query.split("."))
+        except (IndexError, KeyError) as e:
+            raise_from(
+                exceptions.KeySearchNotFoundError(
+                    "Error searching for key in given data"
+                ),
+                e,
+            )
+
+        # If we found a key using 'old' style searching, which will be deprecated
+        warnings.warn(msg, FutureWarning)
```

Comment:
Perhaps we can call warning.warn inside the `_recurse_access_key` function because that will be cleaner when we remove it?

Reply:
Would that not spam the warning a lot of times?

This function will probably be completely removed when we do switch to just using jmespath anyway